mypy/main.py

```diff
@@ -45,6 +45,37 @@ def stat_proxy(path: str) -> os.stat_result:
         return st
 
 
+if sys.platform == 'win32':
+    def python_executable_prefix(v: str) -> List[str]:
+        return ['py', '-{}'.format(v)]
+else:
+    def python_executable_prefix(v: str) -> List[str]:
+        return ['python{}'.format(v)]
+
+
+def _python_version_from_executable(python_executable: str) -> Tuple[int, int]:
+    try:
+        check = subprocess.check_output([python_executable, '-c',
+                                         'import sys; print(repr(sys.version_info[:2]))'],
+                                        stderr=subprocess.STDOUT).decode()
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        return sys.version_info[:2]
```

Comment:
Can you add a comment here explaining why this is the right thing to do?

I don't thing `_python_version_from_executable('bad/path/to/python2.7')` should silently return `"3.6"`...